### PR TITLE
Task #1810: Reset password APi integration

### PR DESCRIPTION
**Pull request by shreyas1434shinde**

### CODE REVIEW

File: public/locales/en/common.json [hunk 1]
```diff
@@ -165,7 +165,8 @@
     "WELCOME": "Welcome!",
     "PLEASE_RESET_YOUR_PASSWORD": "Please reset your password to ensure your account is secure",
     "DO_IT_LATER": "I’ll do it later",
-    "SUCCESSFULLY_RESET": "Your password has been successfully reset"
+    "SUCCESSFULLY_RESET": "Your password has been successfully reset",
+    "INVALID_OR_EXPIRED_TOKEN":"Invalid or Expired Token"
```
Comment (itsvick) [inline public/locales/en/common.json:169]: @shreyas1434shinde change key and value... say link expired

File: src/pages/create-password.tsx [hunk 2]
```diff
@@ -34,6 +38,22 @@ const CreatePassword = () => {
 
   const editPassword = router.pathname.includes('/edit-password');
 
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.localStorage) {
+      const token = localStorage.getItem('token');
+      const role = localStorage.getItem('role');
+      setRole(role);
+      const storedUserId = localStorage.getItem('userId');
+      setClassId(localStorage.getItem('classId') ?? '');
```
Comment (itsvick) [inline src/pages/create-password.tsx:47]: @shreyas1434shinde why classId required here?

Comment (itsvick) [reply src/pages/create-password.tsx:47]: remove unwanted above lines

File: src/pages/reset-password/index.tsx [hunk 3]
```diff
@@ -8,18 +8,36 @@ import { showToastMessage } from '@/components/Toastify';
...
   const { token } = router.query;
 
+  useEffect(() => {
+    if (token) {
+      try {
+        const decodedToken: any = jwt_decode(token as string);
+        const currentTime = Date.now() / 1000;
+        if (decodedToken.exp < currentTime) {
+          setIsTokenValid(false);
+          showToastMessage('Token has expired', 'error');
```
Comment (itsvick) [inline src/pages/reset-password/index.tsx:28]: @shreyas1434shinde translate message

File: src/pages/reset-password/index.tsx [hunk 4]
```diff
@@ -8,18 +8,36 @@ import { showToastMessage } from '@/components/Toastify';
...
+        }
+      } catch (error) {
+        console.error('Invalid token:', error);
+        setIsTokenValid(false);
+        showToastMessage('Invalid token', 'error');
```
Comment (itsvick) [inline src/pages/reset-password/index.tsx:33]: @shreyas1434shinde translate

File: public/locales/en/common.json (hunk 1 above)
Comment (shreyas1434shinde) [reply public/locales/en/common.json]: @itsvick resolved

File: src/pages/reset-password/index.tsx (hunk 3 above)
Comment (shreyas1434shinde) [reply src/pages/reset-password/index.tsx]: @itsvick resolved

File: src/pages/reset-password/index.tsx (hunk 4 above)
Comment (shreyas1434shinde) [reply src/pages/reset-password/index.tsx]: @itsvick resolved

**Pull request merged**